### PR TITLE
[feature] Add location information when throwing XPathException

### PR DESCRIPTION
**Pull request by chris-evolvedbinary**
### Description:

Add location information for all XPathException instances.

### Reference:

https://github.com/eXist-db/exist/issues/3241

### Type of tests:

Existing tests still work and retain location information.

---

This open source contribution to the eXist-db project was commissioned by the Office of the Historian, U.S. Department of State, https://history.state.gov/.

### CODE REVIEW

File: exist-core/src/main/java/org/exist/xquery/XPathException.java [hunk 1]
```diff
@@ -153,30 +151,24 @@ public XPathException(XQueryAST ast, ErrorCode errorCode, String message) {
...
-        }
-    }
-
     /**
      * @param cause the cause throwable
      * @param message the error message
      * @deprecated Use a constructor with errorCode
      */
     @Deprecated
-    public XPathException(final String message, final Throwable cause) {
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/xquery/XPathException.java:174]: I expect incompatibilities with XAR files

File: exist-core/src/main/java/org/exist/xquery/XPathUtil.java [hunk 2]
```diff
@@ -77,27 +78,26 @@ public final static Sequence javaObjectToXPath(Object obj, XQueryContext context
         } else if (obj instanceof Sequence) {
             return (Sequence) obj;
         } else if (obj instanceof String) {
-            final StringValue v = new StringValue((String) obj);
+            final StringValue v = new StringValue(null, (String) obj);
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/xquery/XPathUtil.java:81]: is this really needed this way? cant we use the single argument constructor here?

File: exist-core/src/main/java/org/exist/xquery/XPathException.java [hunk 3]
```diff
@@ -46,16 +46,6 @@ public class XPathException extends Exception implements XPathErrorProvider {
 
     private Source source = null;
 
-    /**
-     * @param message the error message
-     * @deprecated Use a constructor with errorCode
-     */
-    @Deprecated
```
[the remark this answers is not in the document]
Comment (chris-evolvedbinary) [reply exist-core/src/main/java/org/exist/xquery/XPathException.java]: @dizzzz: I re-added this deprecated constructor.

File: exist-core/src/main/java/org/exist/xquery/XPathException.java (hunk 1 above)
Comment (chris-evolvedbinary) [reply exist-core/src/main/java/org/exist/xquery/XPathException.java:174]: @dizzzz: I re-added this deprecated constructor.

File: exist-core/src/main/java/org/exist/xquery/XPathUtil.java (hunk 2 above)
Comment (chris-evolvedbinary) [reply exist-core/src/main/java/org/exist/xquery/XPathUtil.java]: @dizzzz: I re-added all the expressionless value constructors.

**Review by dizzzz — APPROVED** [review]

**Comment by dizzzz** [comment]
There are probably some javadoc issues

**Comment by dizzzz** [comment]
The javadoc build shows inconsistencies and hence fails

https://github.com/eXist-db/exist/runs/7311793037?check_suite_focus=true

**Comment by chris-evolvedbinary** [comment]
@dizzzz: Thank your catching the Javadoc errors.  I fixed them.

**Comment by dizzzz** [comment]
LGTM!

**Comment by dizzzz** [comment]
Oops demarked it as draft

Ok merge?

**Pull request merged**